src/widgets/helper/NotebookTab.cpp

```diff
@@ -381,6 +383,65 @@ void NotebookTab::setHighlightState(HighlightState newHighlightStyle)
     this->update();
 }
 
+void NotebookTab::setHighlightState(HighlightState newHighlightStyle,
+                                    ChannelView &channelViewSource,
+                                    MessagePtr message)
+{
+    if (this->isSelected())
+    {
+        return;
+    }
+
+    if (!this->highlightEnabled_ &&
+        newHighlightStyle == HighlightState::NewMessage)
+    {
+        return;
+    }
+
+    if (this->highlightState_ == newHighlightStyle ||
+        this->highlightState_ == HighlightState::Highlighted)
+    {
+        return;
+    }
+
+    auto splitContainer =
+        dynamic_cast<SplitContainer *>(this->notebook_->getSelectedPage());
+    if (splitContainer != nullptr)
+    {
+        const auto &splits = splitContainer->getSplits();
+        for (const auto &split : splits)
+        {
+            auto &&filterIdsSource = channelViewSource.getFilterIds();
+            auto uniqueFilterIdsSource =
+                QSet(filterIdsSource.cbegin(), filterIdsSource.cend());
+            auto &&filterIdsSplit = split->getChannelView().getFilterIds();
+            auto uniqueFilterIdsSplit =
+                QSet(filterIdsSplit.cbegin(), filterIdsSplit.cend());
+
+            auto isSubset = []<typename T>(QSet<T> sub, QSet<T> super) {
+                for (auto &&subItem : sub)
+                {
+                    if (!super.contains(subItem))
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            };
+
+            if (channelViewSource.underlyingChannel() == split->getChannel() &&
+                split->getChannelView().shouldIncludeMessage(message) &&
+                isSubset(uniqueFilterIdsSource, uniqueFilterIdsSplit))
```

Comment:
We check if the channels are same &&
if message got shown in currently viewed notebook tab splits &&
if source filters (i.e. other tabs split) is subset of current filters (i.e. we only dont-higlight when our current view is "lesser", i.e. more specific))